frontend/src/pages/set_components/settings_components.js

```diff
@@ -221,47 +207,22 @@ class SettingsComponents extends React.Component {
         const currCompName = currState.currComponentName;
         const finalOutput = JSON.parse(localStorage.getItem("fullComponentsInfo"));
         const finalOutputComps = finalOutput.configuration['1'].components;
-        let comp_index = 0;
-        finalOutputComps.map(v => {
+        finalOutputComps.forEach(v => {
             if (v.name === currCompName) {
                 v.parameter = gridRows;
                 //this.setState({"currentParameters": v.parameter});
                 localStorage.setItem("currentParameters", JSON.stringify(v.parameter));
-                let parameters = v.parameter;
-
-                let param_index = 0;
-                v.parameter.map(dependentParameter => {
-                    if (dependentParameter.type === "dependent") {
-                        let match = dependentParameter.parameter.match(/(.*?)--(.*?)--(.*)/);
-                        try {
-                            const dependentParameterOriginalName = match['1'];
-                            const dependentParameterName = match[2];
-                            const dependentParameterNodePath = match[3];
-
-                            v.parameter.map(dynamicParameter => {
-                                if (dependentParameterName === dynamicParameter.parameter && dynamicParameter.type !== "dependent") {
-
-                                    const newSource = dynamicParameter.value;
-                                    const apiRequest = {"new_source": newSource, "node_path": dependentParameterNodePath};
-                                    const response = this.getValueFromSource(apiRequest, parameters, param_index, v.name,
-                                        finalOutputComps, finalOutput, gridRows, comp_index);
-                                }
-                            })
-                        }
-                        catch (e) {}
-                    }
-                    param_index++;
+
+                v.parameter.forEach(() => {
```

Comment:
Why the change? Is it equivalent?

Reply:
(It's hard to have an overview, so it probably makes sense what you did, I'm just curious)

Reply:
Yes, I know. The variables are defined, but nowhere used. Therefore, it's possible to remove all those code. Because of the reason, that the "response" varibale is nowhere used, all these staff can be removed.

![image](https://user-images.githubusercontent.com/34272764/79749585-c15bfa00-830f-11ea-9067-15e1429f86d5.png)